canonical/maas-ui
# feat: collapsible side navigation MAASENG-1282

## Done

- Implement collapsible side navigation
- restructure the navigation to use vanilla classes
- highlight parent when any of its children are active
- remove hidden items when collapsed
- reorganise `l-navigation--` color modifier classes and rename to `is-maas-`
  - use `is-maas-` color className in `ThemedRadioButton`
- add global `[`  keyboard shortcut for expanding and collapsing the navigation

this will be followed up by https://warthogs.atlassian.net/browse/MAASENG-1356 to look into allowing for collapsibility across all breakpoints

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps
small screen size
- Verify that you can open and close the navigation and all items visible

medium screen size:
- Check that the navigation expands on hover
- Verify you can pin and unpin the navigation on medium screen size
- Verify that the navigation is displayed correctly and all items visible

Large screen size
- Verify that the navigation is displayed correctly and all items visible

## Fixes

- https://warthogs.atlassian.net/browse/MAASENG-1331
- https://warthogs.atlassian.net/browse/MAASENG-1282 

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

### After
![Kapture 2023-02-13 at 14 24 39](https://user-images.githubusercontent.com/7452681/218484239-51cdc167-f625-4f80-95c1-b7f940d5f8f4.gif)
